cpp11test/src/test-integers.cpp

```diff
@@ -1,6 +1,6 @@
 #include "Rversion.h"
-#include "cpp11/doubles.hpp"
 #include "cpp11/integers.hpp"
+#include "cpp11/doubles.hpp"
```

Comment:
The order of these includes matters, and is related to the new forward declaration of the following in `integers.hpp`.

```cpp
template <>
double r_vector<double>::operator[](const R_xlen_t pos) const;
```

This declaration was previously missing, but is really required, and the fact that it was required was hidden behind the fact that we included `doubles.hpp` first, creating a hidden dependency (since `doubles.hpp` has the implementation of that forward declaration in it).

Swapping the include order revealed the bug, and is better practice in the long run, I think.